src/Modules/OrchardCore.Commerce/Services/ProductService.cs

```diff
@@ -31,7 +35,19 @@ public async Task<IEnumerable<ProductPart>> GetProductsAsync(IEnumerable<string>
             .Select(idx => idx.ContentItemId)
             .Distinct()
             .ToArray();
-        return (await _contentManager.GetAsync(contentItemIds))
-            .Select(item => item.As<ProductPart>());
+
+        var contentItems = await _contentManager.GetAsync(contentItemIds);
+
+        // We need BuildDisplayAsync to fill part.Elements with the fields. We could extract the logic from
+        // BuildDisplayAsync, but we would have to copy almost everything from there.
+        if (contentItems.Any())
```

Comment:
Why is checking `.Any()` necessary? If `contentItems` is empty the `foreach` will just skip anyway.